### PR TITLE
support deny action in authorization policy

### PR DESCRIPTION
**Pull request by yangminzhu**
This PR adds support to the `action` (ALLOW or DENY) field in authorization policy. E2E tests and unit tests are also added.

For #19894

### CODE REVIEW

**Comment by yangminzhu** [comment]
@istio/wg-networking-maintainers-pilot Can someone review the changes in pilot/pkg/model/authorization.go and pilot/pkg/networking/plugin/authz/authorization.go? We don't need to block on the review of other security related changes. Thanks.

File: pilot/pkg/security/authz/builder/builder_test.go [hunk 1]
```diff
@@ -103,64 +105,84 @@ func TestBuilder_BuildHTTPFilter(t *testing.T) {
...
 					}
-				} else {
-					rbacConfig := &http_config.RBAC{}
-
-					// nolint: staticcheck
-					if got.GetConfig() == nil {
-						t.Errorf("want struct config when isXDSMarshalingToAnyEnabled is false")
-					} else if err := conversion.StructToMessage(got.GetConfig(), rbacConfig); err != nil {
-						t.Errorf("failed to convert struct to message: %s", err)
+					if tc.isXDSMarshalingToAnyEnabled {
```
Comment (howardjohn) [inline pilot/pkg/security/authz/builder/builder_test.go]: outside scope of this pr, but we should remove this, and always us Any

**Pull request merged**